core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1819,17 +1706,12 @@ private CompletableFuture<Integer> produceMessages(int messageCount) {
                 recordFutures[i] = producer.send(record);
             }
             for (int i = 0; i < messageCount; i++) {
-                try {
-                    recordFutures[i].get();
-                    messagesSent++;
-                } catch (Exception e) {
-                    fail("Failed to send record: " + e);
-                }
+                final int index = i;
+                assertDoesNotThrow(() -> recordFutures[index].get());
```

Comment:
Here, can we add the custom error message("Failed to send record") to the assert as well.